api/mfa.go

```diff
@@ -51,3 +58,45 @@ func (a *API) DisableMFA(w http.ResponseWriter, r *http.Request) error {
 	}
 	return sendJSON(w, http.StatusOK, user)
 }
+
+func (a *API) GenerateRecoveryCodes(w http.ResponseWriter, r *http.Request) error {
+	const NUM_RECOVERY_CODES = 8
+	const RECOVERY_CODE_LENGTH = 8
+	ctx := r.Context()
+	user := getUser(ctx)
+	instanceID := getInstanceID(ctx)
+	if !user.MFAEnabled {
+		return MFANotEnabledError
+	}
+	now := time.Now()
+	recoveryCodeModels := []*models.RecoveryCode{}
+	var terr error
+	var recoveryCode string
+	var recoveryCodes []string
+	var recoveryCodeModel *models.RecoveryCode
+	for i := 0; i < NUM_RECOVERY_CODES; i++ {
+		recoveryCode = crypto.SecureToken(RECOVERY_CODE_LENGTH)
+		recoveryCodeModel, terr = models.NewRecoveryCode(user, recoveryCode, &now)
+		if terr != nil {
+			return internalServerError("Error creating recovery code").WithInternalError(terr)
+		}
+		recoveryCodes = append(recoveryCodes, recoveryCode)
```

Comment:
This portion is a little iffy, it assumes that the DB write went through -- not sure if it'd make sense to read from db again and return that or return what was generated assuming the DB write was ok

Reply:
yeah good point, i think its safe to make those assumptions as long as:
1. we check if the transaction below errors out 
2. if the transaction completes successfully, we can be sure that all the writes went through